presto-main/src/main/java/com/facebook/presto/operator/GroupedTopNBuilder.java

```diff
@@ -166,20 +171,20 @@ private void processPage(Page newPage, GroupByIdBlock groupIds)
             }
 
             if (rows.size() < topN) {
-                // still have space for the current group
                 Row row = new Row(newPageId, position);
-                rows.enqueue(row);
                 newPageReference.reference(row);
+                rows.enqueue(row);
```

Comment:
Do we need these changes?

Reply:
The intention behind this change is to update `Row[] PageReference#references` (which should be accessed in a more predictable linear fashion through loop) _before_ enqueueing the `Row` into any given `RowHeap` which is much more likely to incur cache misses and stalls. This is again, probably not easy to measure in isolation- but it also matches the order of operations in the `else` block with the row reference being stored before the `RowHeap#enqueue`